ssz/sedes/serializable.py

```diff
@@ -40,12 +36,34 @@
 TSerializable = TypeVar("TSerializable", bound="BaseSerializable")
 
 
+class Field(abc.ABC):
```

Comment:
If `Field` was done as `class Field(typing.Generic[TValue])` then I think we could get a lot of the type stuff for free on the individual field types without having to implement as many methods.